### PR TITLE
SUP-13392: Added condtion if bgcolor is different than the default, i…

### PR DESCRIPTION
**Pull request by Adirvkaltura**
…f true - the new color is replacing the transprent areas in the image

### CODE REVIEW

File: infra/media/thumbnailMaker/KImageMagickCropper.php [hunk 1]
```diff
@@ -20,6 +20,10 @@ class KImageMagickCropper extends KBaseCropper
 	 * 
 	 */
 	const CROP_AFTER_RESIZE = 6;
+
+        const DEFAULT_BGCOLOR = '0xffffff';
```
Comment (yossipapi) [inline infra/media/thumbnailMaker/KImageMagickCropper.php]: Fix indent.

File: infra/media/thumbnailMaker/KImageMagickCropper.php [hunk 2]
```diff
@@ -45,8 +49,9 @@ public function __construct($srcPath, $targetPath, $cmdPath = null, $forceJpeg =
 		parent::__construct($srcPath, $targetPath);
 	}
 	
-	protected function getCommand($quality, $cropType, $width = 0, $height = 0, $cropX = 0, $cropY = 0, $cropWidth = 0, $cropHeight = 0, $scaleWidth = 1, $scaleHeight = 1, $bgcolor = 0xffffff, $density = 0, $forceRotation = null, $strip = false)
+	protected function getCommand($quality, $cropType, $width = 0, $height = 0, $cropX = 0, $cropY = 0, $cropWidth = 0, $cropHeight = 0, $scaleWidth = 1, $scaleHeight = 1, $bgcolor = DEFAULT_BGCOLOR , $density = 0, $forceRotation = null, $strip = false)
```
Comment (yossipapi) [inline infra/media/thumbnailMaker/KImageMagickCropper.php]: You forgot the self:: before the const.

File: infra/media/thumbnailMaker/KImageMagickCropper.php [hunk 3]
```diff
@@ -291,10 +296,18 @@ protected function getCommand($quality, $cropType, $width = 0, $height = 0, $cro
 				    break;
 			}
 		}
-		
+
+		if ($bgcolor != DEFAULT_BGCOLOR	)
```
Comment (yossipapi) [inline infra/media/thumbnailMaker/KImageMagickCropper.php]: You forgot the self:: before the const.

File: infra/media/thumbnailMaker/KImageMagickCropper.php [hunk 4]
```diff
@@ -20,6 +20,10 @@ class KImageMagickCropper extends KBaseCropper
 	 *
```
Comment (yossipapi) [inline infra/media/thumbnailMaker/KImageMagickCropper.php]: Why did you change the file permission?

File: infra/media/thumbnailMaker/KImageMagickCropper.php (hunk 1 above)
Comment (Adirvkaltura) [reply infra/media/thumbnailMaker/KImageMagickCropper.php]: Fixed

File: infra/media/thumbnailMaker/KImageMagickCropper.php (hunk 2 above)
Comment (Adirvkaltura) [reply infra/media/thumbnailMaker/KImageMagickCropper.php]: Added

File: infra/media/thumbnailMaker/KImageMagickCropper.php (hunk 4 above)
Comment (Adirvkaltura) [reply infra/media/thumbnailMaker/KImageMagickCropper.php]: Added

File: infra/media/thumbnailMaker/KImageMagickCropper.php (hunk 3 above)
Comment (Adirvkaltura) [reply infra/media/thumbnailMaker/KImageMagickCropper.php]: Added

File: infra/media/thumbnailMaker/KImageMagickCropper.php (hunk 4 above)
Comment (Adirvkaltura) [reply infra/media/thumbnailMaker/KImageMagickCropper.php]: I changed the permission for the file on my machine so I'll be able to save changes, otherwise I wasn't able to edit the code.

Comment (yossipapi) [reply infra/media/thumbnailMaker/KImageMagickCropper.php]: Even if you did that to work, dont commit the file permission change.

Comment (Adirvkaltura) [reply infra/media/thumbnailMaker/KImageMagickCropper.php]: I've tried to restore the permissions to the original. Is it right now?

**Comment by MosheMaorKaltura** [comment]
reviewed
